instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py

```diff
@@ -346,6 +346,21 @@ def test_uninstrument(self):
         spans = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(spans), 0)
 
+    def test_no_op_tracer_provider(self):
+        self._client.get("/foobar")
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 3)
+
+        self.memory_exporter.clear()
+        self._instrumentor.uninstrument()
+
+        tracer_provider = NoOpTracerProvider()
+        self._instrumentor.instrument(tracer_provider=tracer_provider)
```

Comment:
Might want to reset the status of `self._instrumentor` to the original before each test.

Reply:
@lzchen I have updated that test to reset the self._instrumentor to the original as requested.

Reply:
@mspillane I think he meant for each test and in order to do so you can override the `setUp` method on the testcase

Reply:
@mspillane sort of, you uninistrumented it on line 363 so you need that line